services/web/server/src/simcore_service_webserver/studies_dispatcher/_redirects_handlers.py

```diff
@@ -92,61 +151,75 @@ async def wrapper(request: web.Request) -> web.StreamResponse:
-            raise _create_redirect_response_to_error_page(
-                request.app,
-                message=f"Invalid parameters in link: {err.text}",
-                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,  # 422
+        except (
+            ServiceNotFound,
+            FileToLargeError,
+            IncompatibleServiceError,
+            GuestUsersLimitError,
+        ) as err:
+            raise _create_simple_error_redirect(
+                request,
+                err,
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            ) from err
+
+        except (InvalidRedirectionParamsError, web.HTTPUnprocessableEntity) as err:
+            # Validation error in query parameters
+            raise _create_error_redirect_with_logging(
+                request,
+                err,
+                message=user_message(
+                    "The link you provided is invalid because it doesn't contain valid information related to data or a service. "
+                    "Please check the link and make sure it is correct.",
+                    _version=1,
+                ),
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                tip="The link might be corrupted",
             ) from err
 
         except web.HTTPClientError as err:
-            _logger.exception("Client error with status code %d", err.status_code)
-            raise _create_redirect_response_to_error_page(
-                request.app,
-                message=err.reason,
+            raise _create_error_redirect_with_logging(
+                request,
+                err,
+                message="Fatal error while redirecting request",
                 status_code=err.status_code,
+                tip="The link might be corrupted",
             ) from err
 
         except (ValidationError, web.HTTPServerError, Exception) as err:
-            error_code = create_error_code(err)
-
-            user_error_msg = compose_support_error_msg(
-                msg=MSG_UNEXPECTED_DISPATCH_ERROR, error_code=error_code
-            )
-            _logger.exception(
-                **create_troubleshootting_log_kwargs(
-                    user_error_msg,
-                    error=err,
-                    error_code=error_code,
-                    error_context={"request": request},
-                    tip="Unexpected failure while dispatching study",
-                )
-            )
-            raise _create_redirect_response_to_error_page(
-                request.app,
-                message=user_error_msg,
+            raise _create_error_redirect_with_logging(
+                request,
+                err,
+                message=MSG_UNEXPECTED_DISPATCH_ERROR,
                 status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                tip="Unexpected failure while dispatching study",
             ) from err
 
     return wrapper
 
 
 #
-# API Schemas
+# API ScheMAS
```

Comment:
really... should that not go in a _models.py module?

Reply:
haha i do not know how that happened. I will refactor.